roles/dlrn_report/tasks/main.yml

```diff
@@ -14,11 +14,21 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+- name: Warning if DLRN is not running
+  debug:
+  msg: "DLRN will not be executed if it is a testproject, make sure you set cifmw_dlrn_report_force_report to true if you want to report to DLRN"
+  when:
+    - ('testproject' in zuul.project.name) and (cifmw_dlrn_report_force_report | bool)
+
 - name: Only report DLRN results when var is set
-  when: cifmw_dlrn_report_result | bool
   block:
     - name: Install dlrnapi-client
       ansible.builtin.import_tasks: install.yml
 
     - name: Report DLRN results
       ansible.builtin.import_tasks: dlrn_report_results.yml
+  when:
+    - zuul is defined
+    - zuul.project is defined
+    - ('testproject' in zuul.project.name) and not (cifmw_dlrn_report_force_report | bool)
```

Comment:
and in opposition to my previous comment, here there should not be `not` in condition probably...?